Fixing images removing for non-default namespaces with / inside

stepdi opened the pull request:
If your namespace is not on Docker Hub (`your.registry/swebench` for example) `should_clean` function will parse the instance image name wrong. `rsplit` instead of `split` fixes that.

john-b-yang commented:
@stepdi Thanks for the fix! Just to make sure I understand, you're talking about non-Docker containerization options where the pattern is something like `your.registry/swebench/an-image-name:latest`, something like this?

Merged because I think this is what you're referring to, and it works fine with the existing SWE-bench docker images, but if this is incorrect or breaks something, we can revisit this.

stepdi commented:
@john-b-yang Yes it works fine with existing SWE-bench docker images because image names there are like `swebench/an-image-name:latest` - so here `.split("/")` and `.rsplit("/")` works the same way - it is used to extract `an-image-name` (it expects `sweb.eval.<something>` or `sweb.<something>` etc. to make a decision about removing the image after use depending on image type and `--clean` setting).

But if I have SWE-bench images in other registry there will be more than one `/` in a full image name so `.split("/")` would return something like `swebench/sweb.<something>` instead of `sweb.<something>` - and [all the](https://github.com/SWE-bench/SWE-bench/blob/7275c821b8976d0e100b4dc9d42205bcf3f45160/swebench/harness/docker_utils.py#L308) `.startswith()` would not work as expected.
